packages/pay/src/index.ts

```diff
@@ -66,19 +66,22 @@ export interface QuoteOptions {
 /** Parameters of payment execution and the projected outcome of a payment */
 export interface Quote {
   /** How payment completion is ascertained: fixed send amount or fixed delivery amount */
-  paymentType: PaymentType
+  readonly paymentType: PaymentType
   /** Maximum amount that will be sent in source units */
-  maxSourceAmount: PositiveInt
+  readonly maxSourceAmount: bigint
+  readonly _maxSourceAmount: PositiveInt
   /** Minimum amount that will be delivered if the payment fully completes */
-  minDeliveryAmount: Int
+  readonly minDeliveryAmount: bigint
+  readonly _minDeliveryAmount: Int
   /** Discovered maximum packet amount allowed over this payment path */
-  maxPacketAmount: PositiveInt
+  readonly maxPacketAmount: bigint
+  readonly _maxPacketAmount: PositiveInt
```

Comment:
But don't these "private" fields complicate the public API even more? If the consumer provides their own quote or resumes a connection, now they need to provide the `Int` and `bigint`. Also, I get that `readonly` should prevent this, but I think it's unclear that only one of the fields affects the payment.

What about only exposing `bigint`s publicly, but validate and convert them to`Int`s within `pay` or `PaymentSender`?